openbook_auth/views.py

```diff
@@ -384,3 +383,54 @@ def get(self, request, user_username):
             user_serializer = GetUserUserSerializer(user, context={"request": request})
 
         return Response(user_serializer.data, status=status.HTTP_200_OK)
+
+
+class PasswordResetRequest(APIView):
+    def post(self, request):
+        request_data = request.data
+        serializer = RequestPasswordResetSerializer(data=request_data)
+        serializer.is_valid(raise_exception=True)
+
+        data = serializer.validated_data
+        has_username = 'username' in data
+        has_email = 'email' in data
+
+        if not has_email and not has_username:
+            return Response('At least one of email or username is required', status=status.HTTP_400_BAD_REQUEST)
+
+        User = get_user_model()
+        user = None
+        if has_username:
+            username = data.get('username')
+            user = User.get_user_with_username(username)
+
+        if has_email:
+            email = data.get('email')
+            user = User.get_user_with_email(email)
+
+        with transaction.atomic():
+            user.request_password_reset()
+
+        return Response('A password reset link was sent to the email',
+                        status=status.HTTP_200_OK)
+
+
+class PasswordResetVerify(APIView):
+    def post(self, request):
+        request_data = request.data
+        serializer = VerifyPasswordResetSerializer(data=request_data)
+        serializer.is_valid(raise_exception=True)
+
+        data = serializer.validated_data
+        token = data.get('token')
+        email = data.get('email')
+        new_password = data.get('new_password')
+
+        User = get_user_model()
+        user = User.get_user_with_email(email)
+
+        with transaction.atomic():
+            user.verify_password_reset_token(token=token, password=new_password)
+
+        return Response('Password set successfully', status=status.HTTP_200_OK)
```

Comment:
Return `ApiMessageResponse` and translate the responses with `_` so it shows up in the frontend.